ansible_collections/arista/cvp/plugins/module_utils/change_tools.py

```diff
@@ -0,0 +1,657 @@
+            else:
+                cc_list = []
+                if change_id is not None:
+                    for change in change_id:
+                        MODULE_LOGGER.debug('Looking up change: ID: %s', change)
+                        cc_list.append(self.get_change_control(change))
+
+                else:
+                    cc_id_list = self._find_id_by_name(name)
+                    for change in cc_id_list:
+                        MODULE_LOGGER.debug('Found change for search: %s with ID: %s', name, change)
+                        cc_list.append(self.get_change_control(change))
+
+                return changed, {'change_controls:': cc_list}, warnings
+
+        elif state == "remove":
+            MODULE_LOGGER.debug("Deleting change control")
+            if change_id is not None:
+                if name is not None:
+                    warnings.append("Deleting CC IDs takes precedence over deleting named CCs. Only the provided CCids will be deleted")
+                try:
+                    self.__cv_client.api.delete_change_controls(change_id)
+                    MODULE_LOGGER.debug("Successfully deleted: %s", change_id)
+                    changed = True
+                except Exception as e:
+                    self.__ansible.fail_json(msg="{0}".format(e))
+
+                return changed, {'remove': []}, warnings
+
+            elif name is not None:
+                cc_list = self._find_id_by_name(name)
+                if len(cc_list) == 0:
+                    warnings.append("No matching change controls found for %s" % name)
+                    return changed, {'search': name}, warnings
+                elif len(cc_list) > 1:
+                    warnings.append("Multiple changes (%d) found matching name: %s" % (len(cc_list), name))
+                    # Should we hard fail here?
+                    e = "Deleting multiple CCs by name is not supported at this time"
+                    self.__ansible.fail_json(msg="{0}".format(e))
+                    return changed, {'matches': cc_list}, warnings
+                else:
+                    try:
+                        MODULE_LOGGER.debug("Trying to delete: %s", cc_list)
+                        data = self.__cv_client.api.delete_change_controls(cc_list)
+                        changed = True
+                    except Exception as e:
+                        self.__ansible.fail_json(msg="{0}".format(e))
+            else:
+                e = "Unable to delete change control. Change name or change_id(s) must be specified"
+                self.__ansible.fail_json(msg="{0}".format(e))
+
+        elif state == "set":
+            changeControl = CvpChangeControlBuilder()
+            changeControl.add_known_uuid([v[1] for v in self.__cc_index])
+            cc_structure = changeControl.build_cc(change, name)
+
+            try:
+                self.__cv_client.post('/api/resources/changecontrol/v1/ChangeControlConfig', data=cc_structure)
```

Comment:
shouldn't we use the wrapper here which also has version checks in place?
https://github.com/aristanetworks/cvprac/blob/develop/cvprac/cvp_api.py#L3388
`change_control_create_with_custom_stages(custom_cc=cc_structure)`

Reply:
It's only in the dev branch, I'll update to use those when cvprac has them available in a release